projects/libs/core-library-typescript/src/state-storage/state-storage.ts

```diff
@@ -48,11 +64,26 @@ export class StateStorage {
   }
 
   private async pollBlockStorageCallback() {
-    const { blocks } = await this.blockStorage.getBlocks({ lastBlockHeight: this.lastBlockHeight });
+    let blocks: types.GetBlocksOutput;
+    try {
+      blocks = await this.blockStorage.getBlocks({ lastBlockHeight: this.lastBlockHeight });
+    }
+    catch (err) {
+      logger.warn(`could not get blocks while polling, last height: ${this.lastBlockHeight}, ${err}`);
+    }
 
-    // Assuming an ordered list of blocks.
-    for (const block of blocks) {
-      await this.syncNextBlock(block);
+    if (blocks != undefined) {
+      // until we finish syncing all of them, lets stop the polling
+      try {
+        this.stopPolling();
```

Comment:
I think it would make sense to stop polling and then ask for blocks to avoid a situation when we have a race condition because of two `getBlocks` calls.